aten/src/ATen/native/layer_norm.cpp

```diff
@@ -206,6 +206,11 @@ std::tuple<Tensor, Tensor, Tensor> math_native_layer_norm(
   const int normalized_ndim = normalized_shape.size();
   // NOLINTNEXTLINE(bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions)
   const int axis = input_ndim - normalized_ndim;
+
+  // Properly handle zero-size inputs: the view(1, M, -1) call below breaks on this.
+  if (input.numel() == 0) {
+    return std::make_tuple(at::empty_like(input), at::empty_like(input), at::empty_like(input));
```

Comment:
@ezyang using this decomp in FakeTensor unearthed a bug with zero-sized tensor inputs (we never hit it before because this decomp isn't used when running directly with cpu or cuda tensors).

From some light testing, it looks like when the input size is zero, the output tensor(s)'s have the same shape as the input. Let me know if this looks reasonable to you.

Reply:
lgtm